src/SonarScanner.MSBuild.Shim/PathHelper.cs

```diff
@@ -64,45 +60,23 @@ public static DirectoryInfo BestCommonRoot(IEnumerable<DirectoryInfo> paths)
             {
                 return null;
             }
-
-            var projectDirectoryParts = paths
-                .Select(GetParts)
-                .ToList();
-
-            var commonParts = projectDirectoryParts
-                .OrderBy(p => p.Count)
-                .First()
-                .TakeWhile((element, index) => projectDirectoryParts.All(p => p[index] == element))
-                .ToArray();
-
-            if (commonParts.Length == 0)
-            {
-                return null;
-            }
-
-            return new DirectoryInfo(Path.Combine(commonParts));
+            var pathParts = paths.Select(GetParts).ToList();
+            var shortest = pathParts.OrderBy(x => x.Length).First();
+            var commonParts = shortest.TakeWhile((x, index) => pathParts.All(parts => parts[index] == x)).ToArray();
+            return commonParts.Length == 0 ? null : new DirectoryInfo(Path.Combine(commonParts));
         }
 
-        public static IList<string> GetParts(DirectoryInfo directoryInfo)
+        public static string[] GetParts(DirectoryInfo directory)
         {
-            if (directoryInfo == null)
-            {
-                throw new ArgumentNullException(nameof(directoryInfo));
-            }
-
+            _ = directory ?? throw new ArgumentNullException(nameof(directory));
             var parts = new List<string>();
-            var currentDirectoryInfo = directoryInfo;
-
-            while (currentDirectoryInfo.Parent != null)
+            while (directory.Parent != null)
             {
-                parts.Add(currentDirectoryInfo.Name);
-                currentDirectoryInfo = currentDirectoryInfo.Parent;
+                parts.Add(directory.Name);
+                directory = directory.Parent;
```

Comment:
I would avoid changing the parameter as a principle

Reply:
In methods like this (single while iterating over parents), we do it on sonar-dotnet too. The rename just adds complexity in tracking why do we need two states. I find this easier to understand